src/Record/Record.Model/FileRecordBuilder.cs

```diff
@@ -81,46 +90,59 @@ this with
         return checksumGraph.Triples;
     }
 
-    private SafeQuad? CreateMediaTypeQuad(string? mediaType) => NullOrDo(mediaType, () => Quad.CreateSafe(_storage.Id!, Namespaces.FileContent.HasMediaType, $"{Namespaces.FileContent.MediaType}{mediaType}", _storage.Id!));
+    private SafeQuad? CreateMediaTypeQuad(string? mediaType) => NullOrDo(mediaType, () => CreateQuadWithPredicateAndObject(Namespaces.FileContent.HasMediaType, $"{Namespaces.FileContent.MediaType}{mediaType}"));
 
-    private SafeQuad? CreateByteSizeQuad(string? byteSize) => NullOrDo(byteSize, () => Quad.CreateSafe(_storage.Id!, Namespaces.FileContent.HasByteSize, $"{byteSize}^^{Namespaces.FileContent.Xsd}decimal", _storage.Id!));
+    private SafeQuad? CreateByteSizeQuad(string? byteSize) => NullOrDo(byteSize, () => CreateQuadWithPredicateAndObject(Namespaces.FileContent.HasByteSize, $"{byteSize}^^{Namespaces.FileContent.Xsd}decimal"));
 
-    private SafeQuad? CreateFileNameQuad(string? fileName) => NullOrDo(fileName, () => Quad.CreateSafe(_storage.Id!, Namespaces.FileContent.HasTitle, fileName, _storage.Id!));
+    private SafeQuad? CreateFileNameQuad(string? fileName) => NullOrDo(fileName, () => CreateQuadWithPredicateAndObject(Namespaces.FileContent.HasTitle, fileName));
 
-    private SafeQuad? CreateLanguageQuad(string? language) => NullOrDo(language, () => Quad.CreateSafe(_storage.Id!, Namespaces.FileContent.HasLanguage, $"{language}^^{Namespaces.FileContent.Xsd}language", _storage.Id!));
+    private SafeQuad? CreateLanguageQuad(string? language) => NullOrDo(language, () => CreateQuadWithPredicateAndObject(Namespaces.FileContent.HasLanguage, $"{language}^^{Namespaces.FileContent.Xsd}language"));
 
     private static SafeQuad? NullOrDo(string? @object, Func<SafeQuad> function) => string.IsNullOrWhiteSpace(@object) ? null : function();
 
 
     #endregion
-    private static byte[] ToByteArray(Stream content)
-    {
-        byte[] contentAsByteArray = new byte[content.Length];
-        content.Read(contentAsByteArray, 0, (int)content.Length);
-        return contentAsByteArray;
-    }
 
-    public IEnumerable<Triple> Build()
+    public Record Build()
     {
-        if (_storage.Id == null) throw new RecordException("File needs the ID from the record of which it will be content for.");
-        if (_storage.Content == null) throw new FileException("File needs content");
-        if (_storage.FileName == null) throw new FileException("File needs a name.");
-        if (_storage.MediaType == null) throw new FileException("File needs a mediatype");
+        if (_storage.Content == null) throw new FileRecordException("File record needs content.");
+        if (_storage.FileName == null) throw new FileRecordException("File record needs a file name.");
+        if (_storage.MediaType == null) throw new FileRecordException("File record needs the mediatype of the file.");
```

Comment:
I think we need checks for _storage.IsSubRecordOf and _storage.Id as well here since they are required.

Reply:
MediaType as well

Reply:
And Langauge

Reply:
Language is not required, and I already check for mediatype. 
It is not necessary to check for _storage.IsSubRecordOf and _storage.Id. This is because the FileRecordBuilder uses the RecordBuilder,  and the RecordBuilder already checks this.